core/entities/variant.go

```diff
@@ -81,7 +76,7 @@ func (v *variant) GetValue(ctx context.Context) string {
 
 func (v *variant) InternalGetValue(ctx context.Context) (returnValue string, isDefault bool) {
 	returnValue, isDefault = v.defaultValue, true
-	mergedContext := context.NewMergedContext(v.globalContext, ctx)
+	mergedContext := context.NewMergedContext(v.parser.GetGlobalContext(), ctx)
```

Comment:
from the next line it seems parser might be nil here..

Reply:
Good point